### PR TITLE
Convert ACT to the new service framework

### PR DESCRIPTION
**Pull request by Cyuubi**
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3521)

### CODE REVIEW

File: src/core/hle/service/act/act.h [hunk 1]
```diff
@@ -4,11 +4,28 @@
 
 #pragma once
 
+#include "core/hle/service/service.h"
+
 namespace Service {
 namespace ACT {
 
+static const u32 MaxACTSessions = 2;
```
Comment (wwylele) [inline src/core/hle/service/act/act.h]: 1. Question: is this reverse engineered from ACT system module or just a place holder? (If it is the latter one, please leave a comment indicating this)
2. If all ports this service opens have the same MaxACTSessions, you don't need to define it here; instead just pass this number directly to `ServiceFramework`'s constructor in `Service::ACT::Module::Interface::Interface`

File: src/core/hle/service/act/act_u.cpp [hunk 2]
```diff
@@ -2,24 +2,19 @@
...
-};
-
-ACT_U::ACT_U() {
-    Register(FunctionTable);
+ACT_U::ACT_U(std::shared_ptr<Module> act)
+    : Module::Interface(std::move(act), "act:u", MaxACTSessions) {
+    static const FunctionInfo functions[] = {
+        {0x00010084, nullptr, "Initialize"},          {0x00020040, nullptr, "GetErrorCode"},
+        {0x000600C2, nullptr, "GetAccountDataBlock"}, {0x000B0042, nullptr, "AcquireEulaList"},
+        {0x000D0040, nullptr, "GenerateUuid"},
```
Comment (wwylele) [inline src/core/hle/service/act/act_u.cpp]: Our convention is to keep one function each line no matter how short it is. Add `clang-format off/on` if needed.

File: src/core/hle/service/act/act.h (hunk 1 above)
Comment (ninjfloof) [reply src/core/hle/service/act/act.h]: Yeah, its a placeholder. Should've clarified that.

File: src/core/hle/service/act/act_u.cpp (hunk 2 above)
Comment (ninjfloof) [reply src/core/hle/service/act/act_u.cpp]: Got it

File: src/core/hle/service/act/act.cpp [hunk 3]
```diff
@@ -10,7 +10,7 @@ namespace Service {
 namespace ACT {
 
 Module::Interface::Interface(std::shared_ptr<Module> act, const char* name, u32 max_session)
```
Comment (wwylele) [inline src/core/hle/service/act/act.cpp]: You aren't using parameter `max_session`. Just remove it.

Comment (Cyuubi) [reply src/core/hle/service/act/act.cpp]: My bad once again.

Comment (Cyuubi) [reply src/core/hle/service/act/act.cpp]: Fixed.

Comment (Cyuubi) [reply src/core/hle/service/act/act.cpp]: ...Actually I broke it whoops.

File: src/core/hle/service/act/act.h [hunk 4]
```diff
@@ -4,11 +4,26 @@
...
 /// Initializes all ACT services
-void Init();
+class Module final {
+public:
+    class Interface : public ServiceFramework<Interface> {
+    public:
+        Interface(std::shared_ptr<Module> act, const char* name);
+        ~Interface();
+
+    protected:
```
Comment (lioncash) [inline src/core/hle/service/act/act.h]: protected is able to be removed.

**Comment by Cyuubi** [comment]
Done.

File: src/core/hle/service/act/act.h (hunk 4 above)
Comment (Cyuubi) [reply src/core/hle/service/act/act.h]: Fixed.

**Pull request merged**